package/gluon-core/luasrc/lib/gluon/upgrade/820-dns-config

```diff
@@ -25,22 +25,33 @@ if dns.servers then
 	})
 end
 
+
+local function set_dns_record(name, host, sectionname)
+		uci:section('dhcp', 'domain', sectionname, {
+			name = name,
+			ip = host,
+		})
+end
+
+-- remove nextnode4_* and nextnode6_*
+uci:foreach("dhcp", "domain", function(s)
+	for key, value in pairs(s) do
+		if (key == ".name" and string.gmatch(value, "nextnode[46].*")) then
+			uci:delete("dhcp", value)
+		end
+	end
+end)
+
 if next_node.name and next_node.ip4 then
-	uci:section('dhcp', 'domain', 'nextnode4', {
-		name = next_node.name,
-		ip = next_node.ip4,
-	})
-else
-	uci:delete('dhcp', 'domain', 'nextnode4')
+	for i,n in ipairs(next_node.name) do
+		set_dns_record(n, next_node.ip4, 'nextnode4_' .. i);
+	end
 end
 
 if next_node.name and next_node.ip6 then
-	uci:section('dhcp', 'domain', 'nextnode6', {
-		name = next_node.name,
-		ip = next_node.ip6,
-	})
-else
-	uci:delete('dhcp', 'domain', 'nextnode6')
+	for i,n in ipairs(next_node.name) do
+		set_dns_record(n, next_node.ip6, 'nextnode6_' .. i);
```

Comment:
semicolon at the end of the line not necessary

Reply:
done